### PR TITLE
feat(ci): create PR bot

### PR DESCRIPTION
**Pull request by smartcontracts**
**Description**
Introduces code for the PR bot. PR bot will leave comments on things and will automatically add/remove labels when necessary.

### CODE REVIEW

File: src/bot.ts [hunk 1]
```diff
@@ -0,0 +1,236 @@
...
+
+import { version } from '../package.json'
+
+type TOptions = {
+  secret: string
+  pat: string
+  tempdir: string
+}
+
+type TMetrics = {}
```
Comment (smartcontracts) [inline src/bot.ts:22]: What metrics would be useful here? We already get server metrics by default.

File: Dockerfile.bot [hunk 2]
```diff
@@ -0,0 +1,14 @@
+FROM node:16-alpine3.14
+
+WORKDIR /app
+
+COPY package*.json /app
+COPY yarn.lock /app
+
+RUN yarn install
```
Comment (roninjin10) [inline Dockerfile.bot:8]:
```suggestion
RUN yarn install --production --frozen-lockfile && yarn cache clean
```

File: package.json [hunk 3]
```diff
@@ -17,16 +17,19 @@
     "test": "jest --detectOpenHandles",
     "lint:check": "eslint . --max-warnings=0",
     "lint:fix": "eslint --fix .",
-    "lint": "yarn lint:fix && yarn lint:check"
+    "lint": "yarn lint:fix && yarn lint:check",
+    "start-bot": "ts-node ./src/bot.ts"
   },
```
Comment (roninjin10) [inline package.json:22]: you could consider adding start-bot to bin https://docs.npmjs.com/cli/v8/configuring-npm/package-json#bin

**Review by roninjin10 — APPROVED** [review]

File: src/bot.ts (hunk 1 above)
Comment (roninjin10) [reply src/bot.ts:22]: I can't think of anything

**Review by mslipper — APPROVED** [review]

**Pull request merged**